crates/miden-block-prover/src/tests/utils.rs

```diff
@@ -140,34 +140,58 @@ pub fn generate_tx_with_authenticated_notes(
     ProvenTransaction::from_executed_transaction_mocked(executed_tx, &chain.latest_block_header())
 }
 
+/// Generates a transaction that expires at the given block number.
+pub fn generate_tx_with_expiration(
+    chain: &mut MockChain,
+    account_id: AccountId,
+    expiration_block: BlockNumber,
+) -> ProvenTransaction {
```

Comment:
Unrelated: what's the reason all these are standalone functions?

Reply:
I was in a rush to finish the tests, so I didn't put too much effort into the test utils. At some point in the middle of it I thought it would be nicer if this was an extension trait but didn't want to spend time on refactoring. Another option is to integrate them into the MockChain directly (but that has to be looked at in isolation and weighed against the overall API of the MockChain) and they're probably too purpose-built for that anyway.

Reply:
I was thinking of the latter, but more in the builder style, e.g.

```rust
chain
  .open_block(..)
    .add_tx()
      .with_account(..)
      .with_expiration(..)
      .finalise()
    .seal();
```

Reply:
Nice idea. I guess this would have "sub-builders" for transactions and batches so that we can craft individual transactions, batches or blocks for whatever is needed for a given test.